File: app/src/main/java/com/cv4j/app/activity/CompareHistActivity.java
package com.cv4j.app.activity;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;

import com.cv4j.app.R;
import com.cv4j.app.adapter.CompareHistAdapter;
import com.cv4j.app.fragment.CompareHist1Fragment;
import com.cv4j.app.fragment.CompareHist2Fragment;
import com.cv4j.app.fragment.CompareHist3Fragment;
import com.safframework.injectview.annotations.InjectExtra;
import com.safframework.injectview.annotations.InjectView;
import com.safframework.injectview.annotations.OnClick;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Tony Shen on 2017/6/4.
 */

public class CompareHistActivity extends BaseActivity {

    @InjectView(R.id.tablayout)
    TabLayout mTabLayout;

    @InjectView(R.id.viewpager)
    ViewPager mViewPager;

    @InjectView(R.id.toolbar)
    Toolbar toolbar;

    @InjectExtra(key = "Title")
    String title;

    private List<Fragment> mList = new ArrayList();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_compare_hist);

        initData();
    }

    private void initData() {

        toolbar.setTitle("< "+title);

        mList.add(new CompareHist1Fragment());
        mList.add(new CompareHist2Fragment());
        mList.add(new CompareHist3Fragment());
        mViewPager.setAdapter(new CompareHistAdapter(this.getSupportFragmentManager(),mList));
        mTabLayout.setupWithViewPager(mViewPager);
    }

    @OnClick(id= R.id.toolbar)
    void clickToolbar() {

        finish();
    }
}

File: app/src/main/java/com/cv4j/app/activity/CompareHistActivity.kt
package com.cv4j.app.activity

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.cv4j.app.R
import com.cv4j.app.adapter.CompareHistAdapter
import com.cv4j.app.app.BaseActivity
import com.cv4j.app.fragment.CompareHist1Fragment
import com.cv4j.app.fragment.CompareHist2Fragment
import com.cv4j.app.fragment.CompareHist3Fragment
import kotlinx.android.synthetic.main.activity_compare_hist.*
import java.util.*

/**
 *
 * @FileName:
 *          com.cv4j.app.activity.CompareHistActivity
 * @author: Tony Shen
 * @date: 2020-05-05 11:40
 * @version: V1.0 <描述当前版本功能>
 */

class CompareHistActivity : BaseActivity() {

    var title: String? = null
    private val mList: MutableList<Fragment> = ArrayList<Fragment>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_compare_hist)

        intent.extras?.let {
            title = it.getString("Title","")
        }?:{
            title = ""
        }()

        toolbar.setOnClickListener {
            finish()
        }

        initData()
    }

    private fun initData() {
        toolbar.title = "< $title"
        mList.add(CompareHist1Fragment())
        mList.add(CompareHist2Fragment())
        mList.add(CompareHist3Fragment())
        viewpager.adapter = CompareHistAdapter(this.supportFragmentManager, mList)
        tablayout.setupWithViewPager(viewpager)
    }
}